File: app/src/main/java/com/example/chatify/ReportPage.java
package com.example.chatify;

import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import java.util.ArrayList;


public class ReportPage extends AppCompatActivity implements MyRecyclerViewAdapter.ItemClickListener {

    MyRecyclerViewAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.report_interface);

        // data to populate the RecyclerView with
        ArrayList<String> reportReaons = new ArrayList<>();
        reportReaons.add("Abusive Speech");
        reportReaons.add("Spam");
        reportReaons.add("Threatening");

        // set up the RecyclerView
        RecyclerView recyclerView = findViewById(R.id.recyclerView);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        adapter = new MyRecyclerViewAdapter(this, reportReaons);
        adapter.setClickListener(this);
        recyclerView.setAdapter(adapter);
    }

    @Override
    public void onItemClick(View view, int position) {

    }
}

File: app/src/main/java/com/example/chatify/ReportPage.kt
package com.example.chatify

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.report_interface.*

class ReportPage : AppCompatActivity() {
    val reasonsList: ArrayList<String> = ArrayList()

    private lateinit var layoutManager: RecyclerView.LayoutManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.report_interface)
        reasonsList.add("Spam")
        reasonsList.add("Abusive Speech")
        reasonsList.add("Threatening a user")
        layoutManager = LinearLayoutManager(this)
        recyclerView.layoutManager = layoutManager
        recyclerView.adapter = RecyclerAdapter(this, reasonsList)
    }
}


